core/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -193,6 +194,13 @@ public RemoteClusterService getRemoteClusterService() {
         return transportService.getRemoteClusterService();
     }
 
+    /**
+     * Return a map of nodeId to pending number of requests for the given action name prefix
+     */
+    public Map<String, Long> getPendingRequests(final String actionNamePrefix) {
```

Comment:
+1 to your suggested patch. I'm just confused why it puts 0 as a default value when it sees a node id for the first time, should it be 1? Similarly it should remove entries from the map in handleResponse when the count is 1 rather than 0?